src/js/node/http2.ts

```diff
@@ -248,31 +351,920 @@ const constants = {
+      if (this.#pingCallbacks) {
+        this.#pingCallbacks.push([callback, Date.now()]);
+      } else {
+        this.#pingCallbacks = [[callback, Date.now()]];
+      }
+    } else if (payload.byteLength !== 8) {
+      const error = new RangeError("ERR_HTTP2_PING_LENGTH: HTTP2 ping payload must be 8 bytes");
+      error.code = "ERR_HTTP2_PING_LENGTH";
+      throw error;
+    }
+
+    parser.ping(payload);
+    return true;
+  }
+  goaway(errorCode, lastStreamId, opaqueData) {
+    return this.#parser?.goaway(errorCode, lastStreamId, opaqueData);
+  }
+
+  setLocalWindowSize(windowSize) {
+    return this.#parser?.setLocalWindowSize(windowSize);
+  }
+  get socket() {
+    const socket = this[bunHTTP2Socket];
+    if (!socket) return null;
+    if (this.#socket_proxy) return this.#socket_proxy;
+    this.#socket_proxy = new Proxy(this, proxySocketHandler);
+    return this.#socket_proxy;
+  }
+  get state() {
+    return this.#parser?.getCurrentState();
+  }
+
+  settings(settings: Settings, callback) {
+    this.#pendingSettingsAck = true;
+    this.#parser?.settings(settings);
+    if (typeof callback === "function") {
+      const start = Date.now();
+      this.once("localSettings", () => {
+        callback(null, this.#localSettings, Date.now() - start);
+      });
+    }
+  }
+
+  constructor(url: string | URL, options?: Http2ConnectOptions) {
+    super();
+
+    if (typeof url === "string") {
+      url = new URL(url);
+    }
+    if (!(url instanceof URL)) {
+      throw new Error("ERR_HTTP2: Invalid URL");
+    }
+    this.#isServer = true;
+    this.#url = url;
+
+    const protocol = url.protocol || options?.protocol || "https:";
+    const port = url.port ? parseInt(url.port, 10) : protocol === "http:" ? 80 : 443;
+
+    // h2 with ALPNProtocols
+    let socket;
+    if (typeof options?.createConnection === "function") {
+      socket = options.createConnection(url, options);
+      this[bunHTTP2Socket] = socket;
+      if (socket.secureConnecting === true) {
+        socket.on("secureConnect", this.#onConnect.bind(this));
+      } else if (socket.connecting === true) {
+        socket.on("connect", this.#onConnect.bind(this));
+      } else {
+        process.nextTick(this.#onConnect.bind(this));
+      }
+    } else {
+      socket = connectWithProtocol(
+        protocol,
+        options
+          ? {
+              host: url.hostname,
+              port,
+              ALPNProtocols: ["h2", "http/1.1"],
+              ...options,
+            }
+          : {
+              host: url.hostname,
+              port,
+              ALPNProtocols: ["h2", "http/1.1"],
+            },
+        this.#onConnect.bind(this),
+      );
+      this[bunHTTP2Socket] = socket;
+    }
+    this.#encrypted = socket instanceof TLSSocket;
+
+    this.#parser = new H2FrameParser({
+      context: this,
+      settings: options,
+      handlers: ClientHttp2Session.#Handlers,
+    });
+
+    socket.on("close", this.#onClose.bind(this));
```

Comment:
Does this code path execute? I'm pretty sure `.bind` is not defined on most of these functions

Reply:
this codepath is the `connect` so should be executing in almost all tests